README.md

```diff
@@ -1,2 +1,285 @@
 # radix-event-stream
-A Rust library to process a stream of events from the Radix Gateway
+
+### An extensible Rust library to help you identify and process events from the Radix ledger.
+
+## Features
+
+### ✅ Extensible:
+
+Easily identify and process custom events by implementing event handlers. Implement custom logging and collect metrics.
+
+### ✅ Data source agnostic:
+
+Pick from one of the provided data sources (Radix Gateway, Radix Gateway database, file) or easily implement your own.
+
+### ✅ Easy to get started with:
+
+Simply pick a transaction source, register your handlers and start the stream.
+
+### ✅ Performant:
+
+Leverages the raw performance of Rust 🦀
+
+### ✅ Async:
+
+Supports asynchronous operations inside handlers for efficient queries.
+
+## Background
+
+[Radix](https://www.radixdlt.com) is a platform for decentralized applications, specifically built for DeFi. Each smart contract, called a component on Radix, can emit events when transactions happen, including custom events defined by the author of the smart contract. An event may look somewhat like this:
+
+```Rust
+#[derive(ScryptoSbor, ScryptoEvent)]
+struct InstantiateEvent {
+    x_address: ResourceAddress,
+    y_address: ResourceAddress,
+    input_fee_rate: Decimal,
+    liquidity_pool_address: ComponentAddress,
+    pool_address: ComponentAddress,
+}
+```
+
+These events are recorded inside of transactions in-sequence and stored on the Radix ledger. This allows developers to track the state of an application by reading events as they happen, and processing them in some specified way. That's what this library aims to achieve.
+
+Events on Radix are encoded as SBOR (Scrypto Binary-friendly Object Representation), a custom format supporting binary and json representations. Radix provides Rust crates which can decode SBOR directly into the events as we defined them in a blueprint. This is possible because Scrypto, the programming language of Radix smart contracts, is based on Rust.
+
+Each event has an **emitter**. This is the on-ledger entity that emitted the event. Events also have a **name**, which is the name of the event and its data type on ledger. These two identifiers are enough to map incoming events to handlers.
+
+## General usage.
+
+Below, the main usage steps are outlined. For more detailed information, see the cargo docs.
+
+### Step 1: copy/paste event definitions.
+
+```Rust
+#[derive(ScryptoSbor, Debug)]
+struct InstantiateEvent {
+    x_address: ResourceAddress,
+    y_address: ResourceAddress,
+    context_fee_rate: Decimal,
```

Comment:
why `context_`?

Reply:
Haha, this just made me realize something.

At some point, I was planning to call one of the parameters to the events "input" instead of "context".
But, I changed my mind, so what I did: I ctrl+f'ed the file and changed "input" to "context". Managed to screw this field up in that action without realizing it 😄 I'll change this back to input.